### ISSUE TITLE
Assignment conversion not possible from type "long" to type "java.lang.Double"

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#2032]
Works: 

```
ksql> SELECT CAST(ROWTIME AS DOUBLE) - CAST(ROWTIME AS DOUBLE) FROM ATM_TXNS_GESS_02 ;
0.0
0.0
```

Doesn't work: 

```
ksql> SELECT CAST((ROWTIME - ROWTIME) AS DOUBLE) FROM ATM_TXNS_GESS_02 ;
Code generation failed for SelectValueMapper
Caused by: Line 1, Column 55: Assignment conversion not possible from type "long" to type "java.lang.Double"
```
